kDrive/UI/View/Files/Preview/CodePreviewCollectionViewCell.swift

```diff
@@ -65,32 +101,37 @@ class CodePreviewCollectionViewCell: PreviewCollectionViewCell {
     }
 
     func configure(with file: File) {
-        do {
-            // Read file
-            let data = try Data(contentsOf: file.localUrl, options: .alwaysMapped)
-            var maybeString: NSString?
-
-            NSString.stringEncoding(for: data, convertedString: &maybeString, usedLossyConversion: nil)
-            guard let maybeString else {
-                throw DriveError.unknownError
-            }
+        let localUrl = file.localUrl
+        let fileId = file.id
 
-            let content = maybeString as String
+        displayLoading()
 
-            // Display content
-            if file.extension == "md" || file.extension == "markdown" {
-                displayMarkdown(for: content)
-                isCode = false
-            } else {
-                displayCode(for: content)
-                isCode = true
+        Task {
+            do {
+                let contentString = try await codePreviewWorker.readDataToStringInferEncoding(localUrl: localUrl)
+                displayContent(with: file, content: contentString)
+            } catch {
+                DDLogError("Failed to read file content: \(error)")
+                previewDelegate?.errorWhilePreviewing(fileId: fileId, error: error)
             }
-        } catch {
-            DDLogError("Failed to read file content: \(error)")
-            previewDelegate?.errorWhilePreviewing(fileId: file.id, error: error)
         }
     }
 
+    private func displayContent(with file: File, content: String) {
+        activityView.stopAnimating()
+        if file.extension == "md" || file.extension == "markdown" {
+            displayMarkdown(for: content)
+            isCode = false
+        } else {
+            displayCode(for: content)
+            isCode = true
+        }
+    }
+
+    private func displayLoading() {
+        activityView.startAnimating()
+    }
```

Comment:
This function only calls `activityView.startAnimating()` and is called once so we don't really need it. Maybe we can simply call
`activityView.startAnimating()`  in the configure ?